tools/device_configuration_checker.py

```diff
@@ -345,4 +349,161 @@ def _run_tests_on_know_good_image(args, device):
         return (result, result_str)
 
     except Exception as error:
+        import traceback
+        traceback.print_exc()
+        logger.error(traceback.format_exc())
         return (False, "Image Test result: " + str(error))
+
+def create_work_directory(directory):
+    """
```

Comment:
Here too, I'm not too fond of the differentiation.
Tbh I do not know much the rest of the implementation, so if this code is aligned with the rest, fine.
However it really starts to look like a perfect case for having a base class and then various children which instantiate/override the mathods that require special handling.